### ISSUE TITLE
ENH: Catch invalid XPath properly when mapping XML

### ISSUE THREAD

**Issue by mielvds** [oeg-upm/morph-kgc#128]
#### Is your Feature Request Related to a Problem?

When your RML file contains invalid XPath, `morph-kgc` crashes.

An example:

```
elem = <Element '{http://www.loc.gov/premis/v3}premis' at 0x1377942c0>, path = 'premis:object/premis:objectIdentifier/premis:objectIdentifierValue/text()'
namespaces = {'premis': 'http://www.loc.gov/premis/v3', 'xsi': 'http://www.w3.org/2001/XMLSchema-instance'}

    def iterfind(elem, path, namespaces=None):
        # compile selector pattern
        if path[-1:] == "/":
            path = path + "*" # implicit all (FIXME: keep this?)
    
        cache_key = (path,)
        if namespaces:
            cache_key += tuple(sorted(namespaces.items()))
    
        try:
            selector = _cache[cache_key]
        except KeyError:
            if len(_cache) > 100:
                _cache.clear()
            if path[:1] == "/":
                raise SyntaxError("cannot use absolute path on element")
            next = iter(xpath_tokenizer(path, namespaces)).__next__
            try:
                token = next()
            except StopIteration:
                return
            selector = []
            while 1:
                try:
>                   selector.append(ops[token[0]](next, token))
E                   KeyError: '()'

../../.pyenv/versions/3.8.7/lib/python3.8/xml/etree/ElementPath.py:368: KeyError
```
Or as mentioned in #121 : absolute paths being used in PO maps instead of paths relative to the iterator.

#### Describe the Solution you'd Like

Catch the error and display a nice error message such as:
- Failed to execute XPath: the Predicate-Object map <http://...> contains absolute paths
- Failed to execute XPath: the template in X contains an invalid XPath path: KeyError: '()'

**Comment by ArenasGuerreroJulian** [comment]
This is related to #63 , I have it in my to-do list.

**Comment by ArenasGuerreroJulian** [comment]
This will be addressed as part of #63
